vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -34,12 +34,12 @@ class Action(VizroBaseModel):
     inputs: List[str] = Field(
         [],
         description="Inputs in the form `<component_id>.<property>` passed to the action function.",
-        regex="^[a-zA-Z0-9_]+[.][a-zA-Z_]+$",
+        regex="^[^.]+[.][^.]+$",
     )
     outputs: List[str] = Field(
         [],
         description="Outputs in the form `<component_id>.<property>` changed by the action function.",
-        regex="^[a-zA-Z0-9_]+[.][a-zA-Z_]+$",
+        regex="^[^.]+[.][^.]+$",
```

Comment:
I don't think this is what we want? I think how we've had it before was safer because it only allowed for alphanumeric characters. In the new regex, something like this would now pass, which didn't pass before: `inputs=["#1.eo"]` 

Or does it not matter at all now?

Reply:
This shouldn't matter now that we always use position. The only limitation on a valid `input` or `output` is that it must contain a single `.` - otherwise anything is allowed.

I don't think there are any restrictions on Dash component IDs (e.g. "1 blah" would work fine) so it should be fine for us also now.

Reply:
So in other words, the components and properties mentioned in `inputs` and `outputs` are still meaningful right? I'm stuck stuck on how this matches with things being positional.

Reply:
Exactly. I tested the graph with the id: `"1scatter #- chart"`, and it works with any predefined or custom action.

Reply:
But this is just for the name of the component targeted right? If this is in first position, it will be passed to the first argument of the function.

We did this, so that we can allow non-pythonic names in components, and pass them positionally to the action functions (same for outputs)?

Reply:
Yes, Max. With the new implementation we can only propagate UI component values as positional so component with any ID can be and input and output for the custom action 😄.